### PR TITLE
Added AVC444 chroma support to shadow server.

### PR DESCRIPTION
**Pull request by akallabeth**
* Added full AVC444 modes to shadow server.
* Primitive always send chroma mode, more bandwidth used but better image quality
* Supports AVC444v1 and AVC444v2

### CODE REVIEW

File: channels/rdpgfx/server/rdpgfx_main.c [hunk 1]
```diff
@@ -508,6 +508,7 @@ static UINT rdpgfx_write_h264_metablock(wStream* s, RDPGFX_H264_METABLOCK* meta)
 	RECTANGLE_16* regionRect;
 	RDPGFX_H264_QUANT_QUALITY* quantQualityVal;
 	UINT error = CHANNEL_RC_OK;
+	Stream_EnsureRemainingCapacity(s, 4 + meta->numRegionRects * 10);
 	Stream_Write_UINT32(s, meta->numRegionRects); /* numRegionRects (4 bytes) */
```
Comment (hardening) [inline channels/rdpgfx/server/rdpgfx_main.c]: Result should be checked

File: channels/rdpgfx/server/rdpgfx_main.c [hunk 2]
```diff
@@ -552,6 +553,7 @@ static INLINE UINT rdpgfx_write_h264_avc420(wStream* s,
 		return error;
 	}
 
+	Stream_EnsureRemainingCapacity(s, havc420->length);
 	Stream_Write(s, havc420->data, havc420->length);
```
Comment (hardening) [inline channels/rdpgfx/server/rdpgfx_main.c]: same here, check result

**Comment by akallabeth** [comment]
@hardening Fixed your comments. Sorry for pushing the wrong commit before that.

**Comment by MartinHaimberger** [comment]
LGTM

**Pull request merged**